mycore-media/src/main/java/org/mycore/media/frontend/jersey/MCRThumbnailResource.java

```diff
@@ -0,0 +1,103 @@
+package org.mycore.media.frontend.jersey;
+
+import org.mycore.access.MCRAccessManager;
+import org.mycore.common.config.MCRConfiguration;
+import org.mycore.datamodel.metadata.MCRMetadataManager;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.datamodel.niofs.MCRPath;
+import org.mycore.frontend.jersey.MCRJerseyUtil;
+import org.mycore.media.services.MCRThumbnailGenerator;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.*;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.nio.file.Files;
+
+import java.nio.file.attribute.FileTime;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Path("thumbnail")
+public class MCRThumbnailResource {
+    @Context
+    private Request request;
+
+    /**
+     * This method returns a thumbnail for a given document with a given size in pixel for the shortest side.
+     * @param documentId the documentID you want the thumbnail from
+     * @param size the size of the shortest side in pixel
+     * @return the thumbnail as png or error 404 if if there is no derivate or no generator for filetype
+     */
+    @GET
+    @Path("{documentId}/{size}")
+    @Produces({"image/png"})
+    public Response getThumbnailFromDocument(@PathParam("documentId") String documentId, @PathParam("size") int size) {
+        return getThumbnail(documentId, size);
+    }
+
+    /**
+     * This method returns a thumbnail for a given document with a default size in pixel for the shortest side.
+     * @param documentId the documentID you want the thumbnail from
+     * @return the thumbnail as png or error 404 if if there is no derivate or no generator for filetype
+     */
+    @GET
+    @Path("{documentId}")
+    @Produces({"image/png"})
+    public Response getThumbnailFromDocument(@PathParam("documentId") String documentId) {
+        int defaultSize = MCRConfiguration.instance().getInt("MCR.Media.Thumbnail.DefaultSize");
+        return getThumbnail(documentId, defaultSize);
+    }
+
+
+    private Response getThumbnail(String documentId, int size) {
+        List<MCRObjectID> derivateIds = MCRMetadataManager.getDerivateIds(MCRJerseyUtil.getID(documentId),
+            0, TimeUnit.SECONDS);
```

Comment:
disables any caching. As this resource is not critical we could live with out-dated data here